rustls/src/client/tls12.rs

```diff
@@ -388,7 +388,10 @@ impl State<ClientConnectionData> for ExpectServerKx {
 
         // Save the signature and signed parameters for later verification.
         let mut kx_params = Vec::new();
-        ecdhe.params.encode(&mut kx_params);
+        ecdhe
```

Comment:
The proliferation of `unwrap()` calls seems like a substantial downside to this approach, I don't think that's the right way to do it.

Reply:
I pushed some changes to avoid adding those extra `unwrap()`s